plugins/feeds/public/cisa_kev.py

```diff
@@ -0,0 +1,137 @@
+
+def _extract_cvss_metric(cve):
+    metrics = cve.get('metrics', {})
+    metric_version = 0
+    for metric in metrics:
+        version = float('.'.join(list(metric.replace('cvssMetricV', ''))))
+        if version > metric_version:
+            metric_version = version
+    if metric_version == 0:
+        return 0, {}
+    metric_str = str(metric_version).replace('.', '')
+    cvss_metric = metrics.get(f'cvssMetricV{metric_str}', [])
+    if len(cvss_metric):
+        return metric_version, cvss_metric[0]
+    else:
+        return 0, {}
+
+class CisaKEV(task.FeedTask):
+    _defaults = {
+        "frequency": timedelta(hours=1),
+        "name": "CisaKEV",
+        "description": "Imports the list of of known exploited vulnerablities published by CISA",
+        "source": "https://www.cisa.gov/known-exploited-vulnerabilities-catalog",
+    }
+
+    CISA_SOURCE: ClassVar["str"] = "https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json"
+    NVD_SOURCE: ClassVar["str"] = "https://services.nvd.nist.gov/rest/json/cves/2.0?hasKev"
+
+    def run(self):
+        response = self._make_request(self.CISA_SOURCE)
+        if not response:
+            logging.warning(f"Unable to fetch feed from {self.CISA_SOURCE}")
+            return
+        kev_json = response.json()
+        response = self._make_request(self.NVD_SOURCE)
+        if not response:
+            logging.warning(f"Unable to fetch feed from {self.NVD_SOURCE}")
+            nvd_json = {}
+        else:
+            nvd_json = _cves_as_dict(response.json())
+        for entry in kev_json.get('vulnerabilities', list()):
+            cve_id = entry.get('cveID')
+            cve_details = nvd_json.get(cve_id, None)
+            self.analyze_entry(entry, cve_details)
+
+
+    def _analyze_cve_details(self, cve_details: str):
+        """Analyzes an entry as specified in nist nvd json."""
+        if cve_details is None:
+            return 0, 'none', ''
+        cve = cve_details.get('cve')
+        cvss_version, cvss_metric = _extract_cvss_metric(cve)
+        if cvss_version == 0:
+            return 0, 'none', ''
+        cvss_data = cvss_metric.get('cvssData', {})
+        base_score = cvss_data.get('baseScore', 0)
+        description = "#### Details\n\n"
+        for cve_description in cve.get('descriptions', []):
+            if cve_description.get('lang') == 'en':
+                description += cve_description.get('value') + '\n'
+        description += f"#### CVSS {cvss_version} Severity and Metrics\n\n"
+        description += f"* **Vector:** {cvss_data.get('vectorString')}\n"
+        description += f"* **Impact Score:** {cve.get('impactScore')}\n"
+        description += f"* **Exploitability Score:** {cve.get('exploitabilityScore')}\n"
+        if cvss_version == 2:
+            severity = cvss_metric.get('baseSeverity', 'none').lower()
+            description += f"* **Access Vector (AV):** {cvss_data.get('accessVector').capitalize()}\n"
```

Comment:
If the key doesn't exist in the dict (which is presumably why you're using `get`), you'll get an attribute error:

```
>>> {}.get('asd').capitalize()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'capitalize'
```

Reply:
Fixed!